File: HotDesk_PRAI/Library/Library.Root/Control/Convertion.vb
Imports System.Web.Script.Serialization

Namespace Control
    Public Class Convertion(Of T)
        Private Shared _ser As JavaScriptSerializer

        ''' <summary>
        ''' Convert List of T into String Format
        ''' </summary>
        ''' <param name="list"></param>
        Public Shared Function Serializer(ByVal list As List(Of T)) As String
            _ser = New JavaScriptSerializer
            Serializer = _ser.Serialize(list)
            _ser = Nothing
        End Function

        ''' <summary>
        ''' Convert string into List of T
        ''' </summary>
        ''' <param name="StringFormat"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Shared Function Deserializer(ByVal StringFormat As String) As List(Of T)
            _ser = New JavaScriptSerializer
            Deserializer = _ser.Deserialize(Of List(Of T))(StringFormat)
            _ser = Nothing
        End Function
    End Class
End Namespace

File: HotDesk_PRAI/Library/Library.Root/Control/Convertion.cs
using System.Web.Script.Serialization;
using System.Collections.Generic;

namespace Control
{
    public class Convertion<T>
    {
        private static JavaScriptSerializer _ser;

        /// <summary>
        /// Convert List of T into String Format
        /// </summary>
        /// <param name="list"></param>
        public static string Serializer(List<T> list)
        {
            _ser = new JavaScriptSerializer();
            string result = _ser.Serialize(list);
            _ser = null;
            return result;
        }

        /// <summary>
        /// Convert string into List of T
        /// </summary>
        /// <param name="stringFormat"></param>
        /// <returns></returns>
        public static List<T> Deserializer(string stringFormat)
        {
            _ser = new JavaScriptSerializer();
            List<T> result = _ser.Deserialize<List<T>>(stringFormat);
            _ser = null;
            return result;
        }
    }
}
